File: Backend/AstroGame.Shared/Models/Buildings/BuildingConstruction.cs
using AstroGame.Shared.Models.Stellar.BaseTypes;
using AstroGame.Shared.Models.Technologies;
using System;

namespace AstroGame.Shared.Models.Buildings
{
    public class BuildingConstruction : TechnologyUpgrade
    {
        public Guid BuildingChainId { get; set; }
        public Guid StellarObjectId { get; set; }

        public virtual BuildingChain BuildingChain { get; set; }
        public virtual StellarObject StellarObject { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/buildings/building_construction.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/technologies/technology_upgrade.dart';

import 'package:json_annotation/json_annotation.dart';

part 'building_construction.g.dart';

@GuidConverter()
@JsonSerializable()
class BuildingConstruction extends TechnologyUpgrade {
  late Guid buildingChainId;
  late Guid stellarObjectId;

  BuildingConstruction();

  factory BuildingConstruction.fromJson(Map<String, dynamic> json) =>
      _$BuildingConstructionFromJson(json);
  Map<String, dynamic> toJson() => _$BuildingConstructionToJson(this);
}
